### PR TITLE
DE-278 | ANALYZE TABLE Functionality

### PR DESCRIPTION
**Pull request by Geoiv**
### Added
- Functionality to perform `ANALYZE TABLE` commands, including on partitions
and on columns. This can be used to increase query performance on specific tables

### Changed
- Added convenience syntax to `add_partition` for providing schema and
table name in the same argument (was missing previously)
- Moved functionality for getting secrets from SSM (AWS parameter store)
from the Salesforce module to its own module

### CODE REVIEW

**Review by jbarno — CHANGES_REQUESTED** [review]
this is neat! Lots of good data will come from analysis.py!

One nitpick/double check on what is arguably a bug by design in python.

File: honeycomb/analysis.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+    if meta.is_partitioned_table(table_name, schema):
+        raise TypeError((
+            'The table {}.{} is partitioned. Use the '
+            '"analyze_partitions" function instead.'
+        ).format(schema, table_name))
+    build_and_run_analysis_command(table_name, schema)
+
+
+def analyze_columns(table_name, schema=None, columns=[]):
```
[the remark this answers is not in the document]
Comment (Geoiv) [reply honeycomb/analysis.py:70]: Ah, that is an interesting point! I have not encountered this specific side effect, but having had other default argument issues with Python, I actually think I know exactly what you're talking about here. I agree that I don't think I'd have any problems here, but I will change it just as a matter of good practice. To be honest, I felt weird when I put a list/dict as a default arg, but I brushed it off, should have paid closer attention to that feeling

**Review by jbarno — APPROVED** [review]
👍

**Review by frankiezeager — APPROVED** [review]
Looks great!

**Pull request merged**